### PR TITLE
test state run requests

### LINKED ISSUE #16901: Request for state run from the minion
The idea here is to allow for the master to send a minion a state run but not run it, instead run it with test=True, and save the output data and the compiled data. Then the state run could be executed by locally via salt-call at the local admin's digression.

We will also need to allow the local admin to view the proposed states and the expected changes, re-run with test=True and exclude any IDs or names in the accrual application of the states.

### PR DESCRIPTION
**Pull request by jfindlay**
closes #16901

### CODE REVIEW

File: salt/modules/state.py [hunk 1]
```diff
@@ -285,7 +293,6 @@ def request(mods=None,
         salt '*' state.request test
         salt '*' state.request test,pkgs
     '''
-    kwargs['test'] = True
```
Comment (cachedout) [inline salt/modules/state.py]: @jfindlay Why was this line removed? This change means that a state will run when it is requested, which I don't believe was the intention here.

Comment (jfindlay) [inline salt/modules/state.py]: I removed it at the request of @thatch45, who was reviewing the code with me as I was trying to get it to work for the integration tests.

Comment (cachedout) [inline salt/modules/state.py]: OK. I will have to ask him about it. I suspect this is a regression but he also wrote it originally so I'm a bit confused. Thanks for the reply.
